cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -829,6 +778,15 @@ determineEra cModeParams localNodeConnInfo =
         Left acqFail -> left $ ShelleyQueryCmdAcquireFailure acqFail
         Right anyCarEra -> return anyCarEra
 
+determineEra2 :: Monad m
+  => ConsensusModeParams mode
+  -> ExceptT e (LocalStateQueryExpr block point (QueryInMode mode) r m) AnyCardanoEra
+determineEra2 cModeParams =
```

Comment:
Maybe rename this to `determineEraExpr`?

Reply:
I was inspired by your suggestion to also rename `sendMsgQuery` to `queryExpr`

Reply:
Also moved `determineEraExpr` to the `Monad` module because it seems pretty general.

Reply:
We should move this to `cardano-api` and call it `eraQueryExpression` or something like that.